src/main/java/com/amplitude/api/AmplitudeClient.java

```diff
@@ -357,6 +358,10 @@ public void onFinished() {
                     deviceId = initializeDeviceId();
                     deviceInfo.prefetch();
 
+                    if (this.deviceIdCallback != null) {
```

Comment:
LGTM, one quick questions is do we just care about the deviceId here? I wonder if we should move this before the `prefetch` method? I guess that method would make calls and causing some latency.

Reply:
Oh yeah, good catch.